<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>949</width>
    <height>387</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Insertion-and-selection-sorts</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_5">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QPushButton" name="push_generate">
          <property name="text">
           <string>Сгенерировать последовательность</string>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="number_for_generate">
          <property name="focusPolicy">
           <enum>Qt::ClickFocus</enum>
          </property>
          <property name="text">
           <string>1000</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="push_selection_sort">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Сортировка выбором</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="push_insertion_sort">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Сортировка включением </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="push_all_sorts">
          <property name="text">
           <string>Обе сортировки</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEdit_5">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="mouseTracking">
           <bool>false</bool>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="acceptDrops">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Время работы сортировки выбором</string>
          </property>
          <property name="frame">
           <bool>false</bool>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="time_selection_sort">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="mouseTracking">
           <bool>true</bool>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="acceptDrops">
           <bool>false</bool>
          </property>
          <property name="frame">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEdit_3">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="mouseTracking">
           <bool>false</bool>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="acceptDrops">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Время работы сортировки включением</string>
          </property>
          <property name="frame">
           <bool>false</bool>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="time_insertion_sort">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="mouseTracking">
           <bool>true</bool>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="acceptDrops">
           <bool>false</bool>
          </property>
          <property name="frame">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QCheckBox" name="push_integer_mode">
            <property name="text">
             <string>Целые</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="push_string_mode">
            <property name="text">
             <string>Строки</string>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <property name="checked">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QLineEdit" name="lineEdit_2">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="mouseTracking">
           <bool>false</bool>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="acceptDrops">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Входные данные</string>
          </property>
          <property name="frame">
           <bool>false</bool>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="input_plain"/>
        </item>
        <item>
         <widget class="QPushButton" name="print_generated">
          <property name="text">
           <string>Напечатать сгенерированные входные данные</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QLineEdit" name="lineEdit">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="mouseTracking">
           <bool>false</bool>
          </property>
          <property name="focusPolicy">
           <enum>Qt::NoFocus</enum>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="acceptDrops">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Сортировка выбором</string>
          </property>
          <property name="frame">
           <bool>false</bool>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="output_plain_selection">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="view_result_selection">
          <property name="text">
           <string>отобразить результат</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <item>
         <widget class="QLineEdit" name="lineEdit_4">
          <property name="mouseTracking">
           <bool>false</bool>
          </property>
          <property name="acceptDrops">
           <bool>false</bool>
          </property>
          <property name="autoFillBackground">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>Сортировка включением</string>
          </property>
          <property name="frame">
           <bool>false</bool>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="output_plain_insertion"/>
        </item>
        <item>
         <widget class="QPushButton" name="view_result_selection_2">
          <property name="text">
           <string>отобразить результат</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>949</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>push_string_mode</sender>
   <signal>clicked()</signal>
   <receiver>push_integer_mode</receiver>
   <slot>toggle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>221</x>
     <y>389</y>
    </hint>
    <hint type="destinationlabel">
     <x>91</x>
     <y>389</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>push_integer_mode</sender>
   <signal>clicked()</signal>
   <receiver>push_string_mode</receiver>
   <slot>toggle()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>93</x>
     <y>399</y>
    </hint>
    <hint type="destinationlabel">
     <x>183</x>
     <y>406</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>push_all_sorts</sender>
   <signal>clicked()</signal>
   <receiver>push_insertion_sort</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>208</x>
     <y>186</y>
    </hint>
    <hint type="destinationlabel">
     <x>163</x>
     <y>145</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>push_all_sorts</sender>
   <signal>clicked()</signal>
   <receiver>push_selection_sort</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>230</x>
     <y>181</y>
    </hint>
    <hint type="destinationlabel">
     <x>231</x>
     <y>127</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
